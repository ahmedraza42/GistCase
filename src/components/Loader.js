import React from 'react'
import { ActivityIndicator, View } from 'react-native'
import { color } from '../utills/colors'

const Loader=({color='#000'})=> {
  return (
    <View style={{ flex: 1, justifyContent: "center", alignItems: "center" }}>
    <ActivityIndicator size={"small"} color={color} />
  </View>
  )
}

export default Loader