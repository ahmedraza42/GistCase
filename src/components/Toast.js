import React from 'react'
import {ToastAndroid, Platform} from 'react-native';
export const showToast = msg => {
  return (
    ToastAndroid.show(msg,ToastAndroid.SHORT)
  )
}

