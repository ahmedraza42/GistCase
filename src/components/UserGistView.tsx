import React from "react";
import { Image, StyleSheet, Text, View } from "react-native";
import { moderateScale } from "react-native-size-matters";
import { color } from "../utills/colors";
import AntDesign from "react-native-vector-icons/AntDesign";
import FontAwesome from "react-native-vector-icons/FontAwesome";
import Feather from "react-native-vector-icons/Feather";

export type Props = {
  userName: string;
  createdAt: string;
  updatedAt: string;
  index: string;
  imagePath: string;
  desc: string;
  fileLength: number;
  fileArray : any[];
};
const UserGistView : React.FC<Props> = ({
  userName,
  createdAt,
  updatedAt,
  index,
  imagePath,
  desc,
  fileLength,
  fileArray = [],
}) => {
  
  const antDesignIcon = (name:string, size:number) => {
    return <AntDesign name={name} color={color.blue} size={size} />;
  };

  return (
    <View key={index?.toString()} style={styles.rootView}>
      <View style={styles.rowCentered}>
        <Image source={{ uri: imagePath }} style={styles.userImage} />
        <Text style={styles.userName}>{userName}</Text>
      </View>

      <View style={{ ...styles.rowCentered, marginTop: moderateScale(5) }}>
        <Text style={styles.semiBoldText}>{createdAt}</Text>
        <Text style={styles.semiBoldText}>{updatedAt}</Text>
      </View>
      <Text style={styles.descText}>{desc || "---"}</Text>

      <View
        style={{
          ...styles.rowCentered,
          ...styles.justifyCenter
        }}
      >
        <View style={{ ...styles.rowCentered }}>
          <Feather name={"code"} color={color.blue} size={moderateScale(15)} />
          <Text
            style={{ ...styles.semiBoldText, ...styles.marginLeft }}
          >{`${fileLength} Files`}</Text>
        </View>
        <View style={{ ...styles.rowCentered }}>
          {antDesignIcon("fork", moderateScale(13))}
          <Text style={{ ...styles.semiBoldText, ...styles.marginLeft }}>
            Forks
          </Text>
        </View>
        <View style={{ ...styles.rowCentered }}>
          <FontAwesome
            name={"comment-o"}
            color={color.blue}
            size={moderateScale(15)}
          />

          <Text style={{ ...styles.semiBoldText, ...styles.marginLeft }}>
            Comments
          </Text>
        </View>
        <View style={{ ...styles.rowCentered }}>
          {antDesignIcon("staro", moderateScale(15))}
          <Text style={{ ...styles.semiBoldText, ...styles.marginLeft }}>
            Stars
          </Text>
        </View>
      </View>

      <View>
        {fileArray.map((item, index) => {
          return (
            <View style={styles.fileArrayRootView}>
              {antDesignIcon("filetext1", moderateScale(15))}
              <Text style={styles.fileNAme}>{item}</Text>
            </View>
          );
        })}
      </View>
    </View>
  );
};
export const MemoizedUserGistView = React.memo(UserGistView);

const styles = StyleSheet.create({
  rootView: {
    width: "99%",
    backgroundColor: "white",
    shadowColor: "rgba(0, 26, 77, 0.7)",
    shadowOffset: { width: 0, height: 0 },
    shadowOpacity: 0.4,
    shadowRadius: 3,
    elevation: 5,
    marginVertical: moderateScale(10),
    borderRadius: moderateScale(12),
    padding: moderateScale(10),
  },
  rowCentered: {
    flexDirection: "row",
    alignItems: "center",
  },
  semiBoldText: {
    fontWeight: "400",
    fontSize: moderateScale(10),
    color: color.text,
  },
  marginLeft: {
    color: color.blue,
    marginLeft: moderateScale(3),
  },
  fileArrayRootView: {
    flexDirection: "row",
    alignItems: "center",
    marginTop: moderateScale(4),
  },
  fileNAme: {
    width: "80%",
    fontSize: moderateScale(10),
    color: color.blue,
    marginLeft: moderateScale(4),
  },
  descText: {
    fontWeight: "400",
    fontSize: moderateScale(11),
    color: color.text,
    marginVertical: moderateScale(8),
  },
  userName: {
    fontWeight: "500",
    fontSize: moderateScale(14),
    marginLeft: moderateScale(10),
    color: color.text,
  },
  userImage: {
    width: moderateScale(30),
    height: moderateScale(30),
    borderRadius: moderateScale(10),
    backgroundColor:'grey'
  },
  justifyCenter:{
    justifyContent: "space-between",
          width: "80%",
  }
});
