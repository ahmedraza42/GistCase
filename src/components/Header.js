import React from "react";
import { StyleSheet, Text, TextInput, View } from "react-native";
import { moderateScale } from "react-native-size-matters";
import AntDesign from "react-native-vector-icons/AntDesign";
import { color } from "../utills/colors";
import Loader from "./Loader";

const Header = ({onChange,value,isLoading=false}) => {
  return (
    <View style={styles.headerRootView}>
      <AntDesign name={"github"} color={color.white} size={moderateScale(30)} />
      <TextInput
        placeholderTextColor={"rgba(41, 38, 42, 0.6)"}
        placeholder="Search Gist for the username"
        multiline
        style={styles.input}
        value={value}
        onChangeText={onChange}
      />
      {isLoading?<Loader color={color.white}/>:null}
    </View>
  );
};

export default Header;

const styles = StyleSheet.create({
  headerRootView: {
    flexDirection: "row",
    padding: moderateScale(10),
    alignItems: "center",
    backgroundColor: color.black,
  },
  input: {
    borderRadius: moderateScale(10),
    height: moderateScale(40),
    backgroundColor: color.white,
    paddingHorizontal: moderateScale(10),
    width: "70%",
    marginLeft: moderateScale(20),
  },
});
