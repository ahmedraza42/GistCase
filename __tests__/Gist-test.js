import React from "react";
import { MemoizedUserGistView } from "../src/components/UserGistView";
import { render, fireEvent } from '@testing-library/react-native';
jest.mock('react-native-size-matters');

test('Gist component snapshot', () => { 
    const { toJSON } = render(<MemoizedUserGistView />);
    expect(toJSON()).toMatchSnapshot();
})


