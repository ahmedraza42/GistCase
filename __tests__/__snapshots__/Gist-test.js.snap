// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gist component snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "borderRadius": undefined,
      "elevation": 5,
      "marginVertical": undefined,
      "padding": undefined,
      "shadowColor": "rgba(0, 26, 77, 0.7)",
      "shadowOffset": {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.4,
      "shadowRadius": 3,
      "width": "99%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        {
          "uri": undefined,
        }
      }
      style={
        {
          "backgroundColor": "grey",
          "borderRadius": undefined,
          "height": undefined,
          "width": undefined,
        }
      }
    />
    <Text
      style={
        {
          "color": "#151515",
          "fontSize": undefined,
          "fontWeight": "500",
          "marginLeft": undefined,
        }
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": undefined,
      }
    }
  >
    <Text
      style={
        {
          "color": "#151515",
          "fontSize": undefined,
          "fontWeight": "400",
        }
      }
    />
    <Text
      style={
        {
          "color": "#151515",
          "fontSize": undefined,
          "fontWeight": "400",
        }
      }
    />
  </View>
  <Text
    style={
      {
        "color": "#151515",
        "fontSize": undefined,
        "fontWeight": "400",
        "marginVertical": undefined,
      }
    }
  >
    ---
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "80%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#2F64ED",
              "fontSize": 12,
            },
            undefined,
            {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          {
            "color": "#2F64ED",
            "fontSize": undefined,
            "fontWeight": "400",
            "marginLeft": undefined,
          }
        }
      >
        undefined Files
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#2F64ED",
              "fontSize": 12,
            },
            undefined,
            {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          {
            "color": "#2F64ED",
            "fontSize": undefined,
            "fontWeight": "400",
            "marginLeft": undefined,
          }
        }
      >
        Forks
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#2F64ED",
              "fontSize": 12,
            },
            undefined,
            {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          {
            "color": "#2F64ED",
            "fontSize": undefined,
            "fontWeight": "400",
            "marginLeft": undefined,
          }
        }
      >
        Comments
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#2F64ED",
              "fontSize": 12,
            },
            undefined,
            {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          {
            "color": "#2F64ED",
            "fontSize": undefined,
            "fontWeight": "400",
            "marginLeft": undefined,
          }
        }
      >
        Stars
      </Text>
    </View>
  </View>
  <View />
</View>
`;
